### PR TITLE
Unlocks fullauto limits, Molly burstfire mode

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Gives molly a 3- and 5-shot rapid burstfire mode instead of fullauto, to differentiate it from the Straylight.

Unlocks full-auto firerate limit, allowing guns to actually vary in firing speed, but reduces the general firerate of all fullauto modes. In effect, this will slow down weapons with very low firerate (such as .40 SMGs), but turn high firerate guns into spray simulator.

Adjusts firerate for tick_lag.

Removes damage debuff from firemodes, significantly buffing all fully automatic and burst weapons. This last change is experimental, and will likely unbalance guns.

Fixes UI related bugs.

## Why It's Good For The Game

Firerate works better, Molly is more unique.

## Changelog
:cl:
balance: gun firerate now more accurate
balance: Molly now has a burst-mode instead of FA
fix: bugs related to firerate and gun UI
/:cl:

### CODE REVIEW

**Comment by tristain01** [comment]
I look forward to this with great interest

File: code/datums/datum_click_handlers.dm [hunk 1]
```diff
@@ -102,8 +103,10 @@
 	object = resolve_world_target(object)
 	if(object)
 		target = object
+		time_since_last_shot = world.time
+		to_chat(owner.mob, "[world.tick_lag], [world.time]")
```
Comment (hyperioo) [inline code/datums/datum_click_handlers.dm:107]: ._.

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -896,8 +896,8 @@
 				"desc" = F.desc,
 				"burst" = F.settings["burst"],
 				"minimum_fire_delay" = GUN_MINIMUM_FIRETIME,
-				"fire_delay" = F.settings["fire_delay"],
-				"move_delay" = F.settings["move_delay"],
+				"fire_delay" = F.settings["fire_delay"] * 5,
+				"move_delay" = F.settings["move_delay"] * 5,
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:900]: Why the `* 5` ?

**Review by hyperioo — CHANGES_REQUESTED** [review]
Small conflict that you can fix with the web interface

File: code/modules/projectiles/gun.dm (hunk 2 above)
Comment (Humonitarian) [reply code/modules/projectiles/gun.dm:900]: It calls ticks ms, but ticks are not real, so I used fortune telling to determine how many ms one tick will be during launch.

**Review by hyperioo — APPROVED** [review]

**Pull request merged**